### PR TITLE
Decouple neoServer creation in InProcessServer

### PR DESCRIPTION
**Pull request by FylmTM**
Main purpose of this change: give possibility to create custom `InProcessServerBuilder` that is capable to use `EnterpriseNeoServer` instead of `CommunityNeoServer`.

---
- Create AbstractInProcessServerBuilder with all necessary functionality
- Add abstract method to create neoServer
- Create CommunityNeoServer in AbstractInProcessServerBuilder
  implementation
- Fix several formatting issues

### CODE REVIEW

File: community/neo4j-harness/src/main/java/org/neo4j/harness/internal/AbstractInProcessServerBuilder.java [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+        }
+        catch ( RuntimeException e )
+        {
+            controls.close();
+            throw e;
+        }
+        return controls;
+    }
+
+    public abstract AbstractNeoServer createNeoServer( Map<String,String> config,
```
Comment (lutovich) [inline community/neo4j-harness/src/main/java/org/neo4j/harness/internal/AbstractInProcessServerBuilder.java]: Probably better make this method `protected`

File: community/neo4j-harness/src/main/java/org/neo4j/harness/internal/InProcessServerBuilder.java [hunk 2]
```diff
@@ -19,48 +19,17 @@
...
-import org.neo4j.server.configuration.ServerSettings;
-
-import static org.neo4j.io.file.Files.createOrOpenAsOuputStream;
-import static org.neo4j.server.configuration.ConfigurationBuilder.ConfiguratorWrappingConfigurationBuilder.toStringForThirdPartyPackageProperty;
-import static org.neo4j.server.configuration.Configurator.DATABASE_LOCATION_PROPERTY_KEY;
-import static org.neo4j.server.configuration.Configurator.WEBSERVER_PORT_PROPERTY_KEY;
-import static org.neo4j.test.Digests.md5Hex;
 
-public class InProcessServerBuilder implements TestServerBuilder
+public class InProcessServerBuilder extends AbstractInProcessServerBuilder
```
Comment (lutovich) [inline community/neo4j-harness/src/main/java/org/neo4j/harness/internal/InProcessServerBuilder.java:31]: `CommunityInProcessServerBuilder` ?

Comment (FylmTM) [inline community/neo4j-harness/src/main/java/org/neo4j/harness/internal/InProcessServerBuilder.java:31]: This will break bacward-compatibility if there is anyone who is extending `InProcessServerBuilder`. If this is OK, then I will rename this class.

**Comment by lutovich** [comment]
@FylmTM thank you for the contribution! This change looks good to me but I have couple questions.
1) What is the intended way of using neo4j-harness after this change?
2) How would we add `Neo4jRule` that uses enterprise database? Currently neo4j-harness lives in community. Would we create a new module with `EnterpriseNeo4jRule`?

@jakewins could you please take a look at this PR?

**Comment by FylmTM** [comment]
Hi @lutovich,

Currently we have `EnterpriseInProcessServerBuilder` with copy-pasted code from `InProcessServerBuilder`, just to change community server to enterprise.

So, after this change we will extened from `AbsractInProcessServerBuilder`, override `createNeoServer` method and create enterprise database there.

We are not using `Neo4jRule`, because we manually handle server (we have a some tweaking for JUnit runners). 

I was thinking about creating `neo4j-harness-enterpise` module, that will provide builder for enterprise server. 

Basically all this is needed because new constraints are only available for enterprise version. We need to test our code against this new constraints, so we need to enterprise database.

File: community/neo4j-harness/src/main/java/org/neo4j/harness/internal/InProcessServerBuilder.java (hunk 2 above)
Comment (lutovich) [inline community/neo4j-harness/src/main/java/org/neo4j/harness/internal/InProcessServerBuilder.java:31]: Right, let's not rename it than.

**Comment by lutovich** [comment]
@FylmTM thanks for explanation. Enterprise equivalent of `Neo4jRule` could probably be done as a separate change.
Let's wait for feedback from @jakewins. He is the initial author of neo4j-harness and would know more about what it is planned to be.

**Comment by jakewins** [comment]
@FylmTM @lutovich LGTM!

**Comment by FylmTM** [comment]
@lutovich Changed public to protected. I can add enterprise version separately in additional pull-request when this one is merged.

**Comment by FylmTM** [comment]
@lutovich Should I review and fix something else?

**Comment by lutovich** [comment]
@FylmTM Everything is good. Sorry I've missed your comment somehow. Thanks a lot for the contribution.
